sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionService.java

```diff
@@ -59,25 +60,32 @@
         "udf.enabled", false,
         Setting.Property.NodeScope), DataTypes.BOOLEAN);
 
+    private final Settings settings;
     private final ClusterService clusterService;
     private Map<String, UDFLanguage> languageRegistry = new HashMap<>();
 
+    @Inject
+    public UserDefinedFunctionService(Settings settings, ClusterService clusterService) {
+        this.settings = settings;
+        this.clusterService = clusterService;
+    }
 
     public UDFLanguage getLanguage(String languageName) throws IllegalArgumentException {
         UDFLanguage lang = languageRegistry.get(languageName);
         if (lang == null) {
-            throw new IllegalArgumentException(String.format(Locale.ENGLISH, "Can't find Language '%s'", languageName));
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH, "'%s' is not a valid UDF language", languageName));
         }
         return lang;
     }
 
-    public void registerLanguage(UDFLanguage language) {
-        languageRegistry.put(language.name(), language);
+    private boolean enabled() {
+        return UDF_SETTING.setting().get(settings);
     }
 
-    @Inject
-    public UserDefinedFunctionService(ClusterService clusterService) {
-        this.clusterService = clusterService;
+    public void registerLanguage(UDFLanguage language) {
+        if (enabled()) {
+            languageRegistry.put(language.name(), language);
+        }
```

Comment:
i think we should log on info level that the language is not registered because udf is disabled